### PR TITLE
search external alias in removed equations, fix init of external vars

### PR DESCRIPTION
**Pull request by vwaurich**
fixes alias detection of external objects and a small fix in initialization of external objects

### CODE REVIEW

**Comment by casella** [comment]
@JKRT I'm not sure @vwaurich has access right to merge. Can you do that for him?

**Comment by JKRT** [comment]
> 
> 
> @JKRT I'm not sure @vwaurich has access right to merge. Can you do that for him?

Yes. I can do a brief review

File: OMCompiler/Compiler/BackEnd/BackendDAECreate.mo [hunk 1]
```diff
@@ -223,14 +226,45 @@ algorithm
...
+  input BackendVarTransform.VariableReplacements repl;
+  output BackendDAE.Variables extVarsOut;
+protected
+  list<BackendDAE.Var> varLst,extVarLst;
+  DAE.ComponentRef cref;
+algorithm
+  varLst := BackendVariable.varList(extVarsIn);
+  extVarLst := {};
+  for var in varLst loop
+		var := matchcontinue(var)
```
Comment (JKRT) [inline OMCompiler/Compiler/BackEnd/BackendDAECreate.mo:254]: I do not understand why matchcontinue is used here, hasReplacement might fail.
Something I have missed?

Also check the indention it is currently not symetric

Comment (kabdelhak) [reply OMCompiler/Compiler/BackEnd/BackendDAECreate.mo:254]: Old coding style, i guess force of habit

**Comment by JKRT** [comment]
@casella  Fixed the indention and replaced the matchcontinue, with match. Merging

**Pull request merged**